.github/workflows/maintenance.yml

```diff
@@ -0,0 +1,52 @@
+name: Maintenance 🧰
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  workflow_removal:
+    runs-on: ubuntu-latest
+    name: Workflow cleanup 🧹👷‍♂️
+    # Errors will probably be caused by excesses in API quota, so we can safely continue.
+    # Remaining workflows will be removed in the next scheduled run.
+    continue-on-error: true 
+    strategy:
+      fail-fast: false
+      matrix:
+        repo:
+          - 'lowlighter/metrics'
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2.3.4
+
+      - name: Run deletion script 🗑
+        run: ./delete_workflows.sh ${{ matrix.repo }}
+        working-directory: ./.github/workflows/maintenance
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
```

Comment:
It isn't sadly, I tested it myself in my personal repo and ``gh`` wasn't happy with it. Take in mind that ``delete:packages`` is an entire seperate scope in PAT. The actions that somebody suggested here (I got the notification in my phone, but it seems he removed his comment), like [this one](https://github.com/marketplace/actions/container-retention-policy) also require a PAT token.

Also what's your stance in using an action for it? I made the script more than 2 months ago and, at that point, I wasn't able to find any action for it, that's why I didn't check prior making this PR. However, the action seem to work differently than my script, removing older tags as well (while my script just removes dangling images). The packages are a pretty sensitive topic, so I myself prefer to have it fully in my control and not rely on an action for that, but maybe your opinion is different. I can switch to an action if you prefer


As for the workflow one, you're right and the repo's token is enough. I originally made this action for running it across multiple repositories, so I needed the token. However, for the repo the action is running, I just tested and it works perfectly. Do you plan on using it for your other repos? If so I can add them to the matrix, if not, I can switch to the default token.

Reply:
> It isn't sadly, I tested it myself in my personal repo and gh wasn't happy with it. Take in mind that delete:packages is an entire seperate scope in PAT

Even if you explicitely grant the repository the `role: write` through the package settings ? (GitHub recently added this which "lowered" rights for `github.token` if you didn't explicitely grants them again)
![image](https://user-images.githubusercontent.com/22963968/140241048-65eb9252-bb56-4c0b-871c-a62db7143257.png)

> Also what's your stance in using an action for it? I made the script more than 2 months ago and, at that point, I wasn't able to find any action for it, that's why I didn't check prior making this PR. However, the action seem to work differently than my script, removing older tags as well (while my script just removes dangling images). The packages are a pretty sensitive topic, so I myself prefer to have it fully in my control and not rely on an action for that, but maybe your opinion is different.

Unless it's offered by GitHub directly (or if the action is related to helping repo management and non-critical stuff like labeling, archiving stuff, etc.) I prefer having scripts upstream, especially when it could affect end-users. 
Since this action is dealing with a lot of PATs it's also more secure for them and feels safer I guess.

> Do you plan on using it for your other repos? If so I can add them to the matrix, if not, I can switch to the default token.

If I reuse it for others repos I don't plan on using workflows from this repo to clean them (it'd feel weird to have a repo messing up with another unlerated repo) so I'd prefer using the default token